staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/equality.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var equalities = func() conversion.Equalities {
+	// Clone equality.Semantic
+	// Add a comparator for managed fields in addition to semantic equality
+	e := conversion.EqualitiesOrDie(
+		func(a, b metav1.ManagedFieldsEntry) bool {
+			// Two objects' managed fields are equivalent if, ignoring timestamp,
+			//	the objects are deeply equal.
+			a.Time = nil
+			b.Time = nil
+
+			// Purposefully using reflect.DeepEqual, not semantic equality.
+			return reflect.DeepEqual(a, b)
+		},
+	)
+	for key, value := range equality.Semantic.Equalities {
+		// Can't call e.AddFunc on value, reflection has type-erased it and
+		// AddFunc throws error. So we directly set the value in the map
+		e.Equalities[key] = value
+	}
+
+	return e
+}()
+
+// IgnoreManagedFieldsTimestampsTransformer reverts any timestamp updates
+//
+func IgnoreManagedFieldsTimestampsTransformer(
+	_ context.Context,
+	newObj runtime.Object,
+	oldObj runtime.Object,
+) (runtime.Object, error) {
+	if equalities.DeepEqual(newObj, oldObj) {
+		// Remove any changed timestamps, so that timestamp is not the only
+		// change seen by etcd.
+		//
+		oldAccessor, err := meta.Accessor(oldObj)
+		if err != nil {
+			return nil, err
```

Comment:
Maybe we want to distinguish the error on the "new" accessor and the "old" accessor.